packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -759,3 +829,176 @@ def test_dcos_net_cluster_identity(dcos_api_session):
     cookie = subprocess.check_output(argv, stderr=subprocess.STDOUT).decode('utf-8').rstrip()
 
     assert cluster_id == cookie, "cluster_id: {}, cookie: {}".format(cluster_id, cookie)
+
+
+class CalicoNetworkTestBase(ABC):
```

Comment:
Thanks for your effort in making our tests stable and robust :)
And we have `test_vip` code for calico container networking, `test_calico_vip`.
For what else is missing is test_vip, can we test applications communication in one node or different node? can we test the container network is in expected network CIDR?
I don't mind removing these new tests only if our new feature is safe.

Reply:
`test_vip` starts containers on same node and on different nodes with and without named vip and checks connectivity between all the pairs on 3 ucr and 1 docker runtimes. it doesn't check which ip addresses are allocated for tasks, but you can easily add such check to `test_vip` if you want. I believe it's just few very simple lines of code.

Reply:
`test_vip` covers too many levels of tests, if we have container communication error , only `test_vip` will be red, and guess what if we want to add a network policy and test the connective between two containers, we may add another `if/else` in `test_vip`?

Reply:
With a standalone file, it's much easier to run e2e test for calico networking as shown in https://github.com/dcos/dcos/blob/d8b21837464c5c1d91dd2f0a582f04ec5b8c35c5/test-e2e/test_calico_networking.py.

test_networking.py has encountered a lot of problems, so let's keep minimal change for it, and suggest only augment changes for it. And for calico networking, as you can see, I added only fundamental and augment tests, and there will be network policy functions in the future. . And I am glad to maintain this part of networking tests as a standalone test file.

Reply:
cc @GoelDeepak

Reply:
Calico network policies is out of scope for this PR. I would move new classes to another branch, focus upon just bringing calico into dc/os and later do some refactoring in tests to support more cases in `test_vip` when it is required. If you insist on having those new tests as a part of this PR, well, it's your choice. I won't be able to review just these classes, but always here if you need any help.

Reply:
There are two testing requirements: 
1. Test the default installation of Calico
2. Test the advance options of Calico

we should use `test_vip` to test 1. The benefit is that `test_vip` would exhaustively verify if the default installation is working. For 2, we could have a separate file. Although it would be nice to have all the networking related tests in `test_networking.py` , however, I don't have a strong opinion about it

Reply:
To make sure we are on the same page.
- we use test_vip to cover the connection test of calico networking, and drop the test what contains in `test_calico_networking.py` except that container IP is in expected CIDR check.
- for advanced option, we can still use test_networking.py and  specifying the test function name to include the related tests.

Thanks @urbanserj  and @GoelDeepak